src/test/java/org/jabref/logic/importer/fetcher/SearchBasedFetcherCapabilityTest.java

```diff
@@ -0,0 +1,48 @@
+package org.jabref.logic.importer.fetcher;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Defines the set of capability tests that each tests a given search capability, e.g. author based search.
```

Comment:
I don't think it's a good strategy to have a test interface. As you have already noted not every fetcher supports all of these fields, and it is also not possible to make sure that the tests defined by this interface are consistent across the fetchers.

What would make more sense is to have tests (against the interface itself) that checks that all implementing classes perform the same under certain conditions (e.g. empty query results in empty result etc). Personally I would restrict those general tests to simple test situations, although it should be possible to use parameterized tests to also check for more complex scenarios.

Reply:
These tests are more low-level. That was tried to be explained there:

> Here, I am systematically evaluating the capabilities of fetchers. This is a) for documentation in code b) to check whether the fetchers change their capability over time. I accept that tests may fail due to changing external services.

"documentation as code"

The screnario explained by you builds on the "results" of these tests. These test ensure that the idea of "your" tests can be implemented.

Reply:
In other words: Instead of documenting the features of a fetcher in markdown, Google Docs, Latex,  PDF or JavaDoc, the documentation is put as code. This ensures that one implementation aspect (query synatx) is checked, changes of the capabilities are detected at the level.of the query and that the list of capabilities (avail/unavail) are noted down as test case.

When documenting in Markdown (or other text), a programmer has to update the documentation in case a capability changes. It is open who enforces this (e.g., do the reviewers know that there is documentation somewhere?) - With the "documentation as code" approach, the programmer is forced to update the documentation

Reply:
I'm in favor of adding these tests, but I don't see the value of the test interface. The interface methods don't say anything about the actual contract that the fetcher has to satisfy. At most they serve as a reminder to test certain aspects of the fetcher.

Reply:
Therefore, more tests covering the contract will come in. These tests are one level above the current ones. This here is like testing Java Byte Code, the next tests will tests Java Code

Reply:
Will be a "research" effort to compare the semantics. E.g. is "year" really the same for each fetcher? Submission date versus published date. - Is "journal" the same thing for all.

We can surely work on that. Then the PR will posaibly be ready in September if all other parts of the Theiss are finished. 

With that, I surely accept that there might be changes in the future . They definitely need to be discussed.

The idea is to have rather small PRs than 10kLOC to review as a whole in September.

Reply:
As I said above, the idea of the tests is good. I also agree with many small PRs instead of a big one.

But can you please explain to me what the advantages of the test interface are? The test interface is never used (only implemented), thus from an abstract object-oriented view it's meaningless, right?

Reply:
Implementation-only is also valid from oo perspective. The alternative would be an abstract class. In the concrete case,  the test classes implementing the interface implement also more tests. The interface just documents that the respective tests are implemented for the concrete fetcher. Documentation as code (answering the question for which fetchers a capability test is implemented)

Reply:
I strongly prefer that the interface adds real tests and not just serves as a reminder. See for example the StringTest at the end of https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-interfaces-and-default-methods. There the ComparableTest interface really defines how the object under test behaves. The implementing classes then only have to provide the correct input/output variables.

For now we can keep the simple interface, but this should be really improved as the project continuous.